axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/SaleOrderInvoiceServiceImpl.java

```diff
@@ -953,6 +953,34 @@ public void displayErrorMessageBtnGenerateInvoice(SaleOrder saleOrder) throws Ax
     }
   }
 
+  @Override
+  public int getInvoicingState(SaleOrder saleOrder) {
+    int invoicingState = 0;
```

Comment:
Please use static variables like we do for selections instead of hardcoded numbers